examples/notebooks/beam-ml/rag_usecase/redis_connector.py

```diff
@@ -0,0 +1,406 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+import logging
+from past.builtins import unicode
+
+import apache_beam as beam
+import numpy as np
+
+from apache_beam.transforms import DoFn
+from apache_beam.transforms import PTransform
+from apache_beam.transforms import Reshuffle
+from apache_beam.utils.annotations import deprecated
+from apache_beam.options.value_provider import ValueProvider
+from apache_beam.options.value_provider import StaticValueProvider
+from apache_beam import coders
+
+import typing
+
+# Set the logging level to reduce verbose information
+import logging
+
+logging.root.setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+
+import redis
+
+__all__ = ['InsertDocInRedis', 'InsertEmbeddingInRedis']
+
+
+class Document(typing.NamedTuple):
+    id: str
+    url: str
+    title: str
+    text: str
+
+
+coders.registry.register_coder(Document, coders.RowCoder)
+
+"""This module implements IO classes to read write data on Redis.
+
+
+Insert Doc in Redis:
+-----------------
+:class:`InsertDocInRedis` is a ``PTransform`` that writes key and values to a
+configured sink, and the write is conducted through a redis pipeline.
+
+The ptransform works by getting the first and second elements from the input,
+this means that inputs like `[k,v]` or `(k,v)` are valid.
+
+Example usage::
+
+  pipeline | WriteToRedis(host='localhost',
+                          port=6379,
+                          batch_size=100)
+
+
+No backward compatibility guarantees. Everything in this module is experimental.
+"""
+
+
+@deprecated(since='v.1')
+class InsertDocInRedis(PTransform):
+    """InsertDocInRedis is a ``PTransform`` that writes a ``PCollection`` of
+    key, value tuple or 2-element array into a redis server.
+    """
+
+    def __init__(self, host=None, port=None, command=None, batch_size=100):
```

Comment:
Instead of doing manual type checking in this function, annotate your input types: https://docs.python.org/3/library/typing.html

You also shouldn't specify `None` base types for required inputs, those are only valid defaults for optional parameters (which should be type hinted as such)

Reply:
Resolved